modules/tests/shared/src/test/scala/coursier/test/VersionTests.scala

```diff
@@ -45,7 +45,9 @@ object VersionTests extends TestSuite {
 
     'buildMetadata - {
       * - {
-        assert(compare("1.2", "1.2+foo") < 0)
+        assert(compare("1.2", "1.2+foo") == 0)
```

Comment:
https://semver.org/spec/v2.0.0.html#spec-item-10 says:

> Build metadata MUST be ignored when determining version precedence. 

so not just vs `1.2+bar`, also vs plain `1.2`, `1.2+foo` should be treated the same.

Reply:
Yes, but that introduces some non-determinism when ordering versions. (The underlying artifacts are likely to be different, so it's better not to randomly choose one of the versions.)

Reply:
The same goes for any of the things that would evaluate to `== 0`, like `ga` and `final`.
Also `2.0+20130313144700` coming after `2.0.2` is a bug, I'd say.